src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeSyntaxNode,
+                SyntaxKind.ExpressionStatement,
+                SyntaxKind.EqualsValueClause,
+                SyntaxKind.Argument,
+                SyntaxKind.ArrowExpressionClause);
+        }
+
+        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        {
+            var parentSyntaxNode = context.Node;
+
+            // Find the invocation expression i.e
+            // a method/delegate call or a property access
+            // or a chain of them
+            var invocationExpressionSyntax = parentSyntaxNode
+                .DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .FirstOrDefault();
+
+            if (invocationExpressionSyntax is null)
+            {
+                // This does not contain any invocations, bail out
+                return;
+            }
+
+            bool IsLambdaExpression(SyntaxNode syntaxNode) => syntaxNode.IsKind(SyntaxKind.SimpleLambdaExpression);
+
+            bool IsNotArgumentSyntax(SyntaxNode syntaxNode) => !syntaxNode.IsKind(SyntaxKind.Argument);
+
+            // Find all non-nested arguments which contain a lambda expression
+            var nonNestedArgumentsWithLambdas = invocationExpressionSyntax
+                .DescendantNodes(IsNotArgumentSyntax)
+                .OfType<ArgumentListSyntax>()
+                .Where(argument => argument.DescendantNodes().Any(IsLambdaExpression));
+
+            // There is none or only a single call
+            if (nonNestedArgumentsWithLambdas.Count() < 2)
+            {
+                // No more refactoring required
+                return;
+            }
+
+            // The first argument can stay on the same line
+            var successiveArgumentsWithLambdas = nonNestedArgumentsWithLambdas.Skip(1);
+
+            foreach (var argument in successiveArgumentsWithLambdas)
+            {
+                // Get the parent invocation expression
+                var parentInvocation = argument.Parent;
+
+                // Find the method/delegate call
+                var memberAccessExpression = parentInvocation?.ChildNodes().OfType<MemberAccessExpressionSyntax>().FirstOrDefault();
+
+                if (memberAccessExpression is null)
+                {
+                    // unknown syntax, don't know what to do
+                    break;
+                }
+
+                // Get the dot operator which is used to call the method/delegate
+                var dotToken = memberAccessExpression.ChildTokens().FirstOrDefault(token => token.IsKind(SyntaxKind.DotToken));
+
+                // If the dot operator does not have a leading whitespace, report violation
+                if (!dotToken.LeadingTrivia.Any(trivia => trivia.IsKind(SyntaxKind.WhitespaceTrivia)))
```

Comment:
So we won't allow short chains on the same line e.g. `values.Where(x => x.Num % 2 == 0).Select(x => x.Num);`?

Reply:
That is my understanding of what we want. I'll loop in @gregr-ni here, but we decided that LINQ statements with lambdas make things harder to read, so always split multiples onto separate lines. It keeps the rules easier to understand as well, rather than also thinking about line length, since we actually have no hard limit on length.

Reply:
What Jayson said. Trying to be more lenient makes it much harder to enforce. This is very similar to requiring all parameters being on separate lines. Yes, there are cases where having more than one on a line would be ok, but it is way simpler to be strict.

Reply:
I wasn't sure if it was noticed, so calling it out explicitly. This analyzer is allowing the first lambda expression to stay on the same line. So this...

```cs
values.Where(x => x.Num % 2 == 0)
   .Select(x => x.Num);
```

...is not going to generate a diagnostic.

Reply:
I think that makes sense

Reply:
We have a fixer. If we really need to change it, we should be able to apply it to everything at once right?

Reply:
> We have a fixer.

We don't have one yet. At-least, not with this PR. Although I am writing one and will post a follow up soon.

> If we really need to change it, we should be able to apply it to everything at once right?

Yes, changing it should be easy.